File: Modules/Help.cs
using System.Text;
using System.Linq;
using SteamKit2;

namespace SteamNerd.Modules
{
    class Help : Module
    {
        public Help(SteamNerd steamNerd) : base(steamNerd)
        {
            Name = "Help";
            Description = "Helps you get help.";

            AddCommand(
                "help",
                string.Format("Helps you get help from Help. Usage: {0}help or {0}help [module]", SteamNerd.CommandChar),
                GetHelp
            );
        }

        public void GetHelp(SteamFriends.ChatMsgCallback callback, string[] args)
        {
            var chat = callback.ChatRoomID;

            if (args.Length == 1)
            {
                var message = "Help is here to help you.\nModules:\n";

                foreach (var module in SteamNerd.Modules)
                {
                    if (module.Name != null && module.Name != "")
                    {
                        message += module.Name + "\n";
                    }
                }

                message += "\nIf you would like more help on a certain module, type {0}help [module]";

                SteamNerd.SendMessage(string.Format(message, SteamNerd.CommandChar), chat, true);
            }
            else
            {
                var modString = args.Skip(1).Aggregate((mod, next) => mod + " " + next);

                var module = SteamNerd.GetModule(modString);

                if (module == null)
                {
                    SteamNerd.SendMessage(string.Format("Module {0} not found!", args[1]), chat, true);
                    return;
                }

                ModuleHelp(module, chat);
            }
        }

        /// <summary>
        /// Prints out help information to a chat
        /// </summary>
        /// <param name="chatRoom"></param>
        public void ModuleHelp(Module module, SteamID chatRoom)
        {
            var message = new StringBuilder(string.Format("{0}\n{1}\n\n", module.Name, module.Description));

            foreach (var command in module.Commands)
            {
                if (command.Match[0] == "" || command.Description == "")
                {
                    continue;
                }

                var match = new StringBuilder();
                
                foreach (var arg in command.Match)
                {
                    match.Append(arg + " ");
                }

                message.Append(string.Format("{0}{1,-50}{2}\n", SteamNerd.CommandChar, match, command.Description));
            }

            SteamNerd.SendMessage(message.ToString(), chatRoom, true);
        }
    }
}

File: Modules/Help.py
Module.Name = "Help"
Module.Description = "Help. When you need it, where you need it."
Module.Global = True

def GetHelp(callback, args):
	if len(args) == 1:
		ListModules(callback)
	else:
		GetHelpOnModule(callback, args) 
	
def ListModules(callback):
	message = "Help is here to help you.\nModules:\n"
	
	for module in SteamNerd.GetModules(callback.ChatRoomID):
		message += module.Name + "\n"
		
	message += "\nIf you would like more help on a certain module, type {}help [module]."
	message = message.format(SteamNerd.CommandChar)
	
	SteamNerd.SendMessage(message, callback.ChatRoomID)
	
def GetHelpOnModule(callback, args):
	moduleName = ' '.join(args[1:])
	module = SteamNerd.GetModule(moduleName, callback.ChatRoomID)
	
	if module == None:
		SteamNerd.SendMessage("{} not found!".format(moduleName), callback.ChatRoomID)
	else:
		message = "{}\n{}\n\n".format(module.Name, module.Description)
		
		for command in module.Commands:
			message += "{}{:<50}{}\n".format(SteamNerd.CommandChar, ' '.join(command.Match), command.Description)
	
		SteamNerd.SendMessage(message, callback.ChatRoomID)
	
Module.AddCommand("help", 
	"Helps you get help from Help. Usage: {0}help or {0}help [module]".format(SteamNerd.CommandChar),
	GetHelp
)